### PR TITLE
bug(modals): Fix Modal padding so Backdrop is clickable on the top edge

### PR DESCRIPTION
**Pull request by ryan953**
Update `<Modal>` dialogs so that user can click above the modal to dismiss it; this is in addition to clicking the Backdrop on the left/right/bottom sides. 

| | Note | Img |
| --- | --- | --- |
| Before | The padding extended 80px to the top of the modal, which positioned things but also captured clicks. Clicking on any of the green areas would not close the modal | before |
| After | The padding extends the same amount on top/bottom, margin is used to maintain position. Clicking outside of the green area will dismiss the modal (clicks in the orange will dismiss the modal). | after |

NB: I went with `margin-top` because `top` itself wasn't working. The parent node, `<Container>`,  is `position: fixed` but the `<Dialog>` itself isn't positioned. I thought that kind of change is more invasive and could create collateral issues.

Looks like the `padding: 80px` went in during https://github.com/getsentry/sentry/pull/25768, I can't tell if it was done for any reason other than alignment.

### CODE REVIEW

File: static/app/views/performance/transactionSummary/transactionThresholdModal.tsx [hunk 1]
```diff
@@ -294,7 +294,6 @@ const Instruction = styled('div')`
 export const modalCss = css`
   width: 100%;
   max-width: 650px;
-  margin: 70px auto;
```
Comment (ryan953) [inline static/app/views/performance/transactionSummary/transactionThresholdModal.tsx:297]: Leaving this alone caused a visual snapshot regression, i pulled it out because it's not really doing much. If there are regressions anyway then i'd be ok with that because the position of this modal is more consistent.

File: static/app/components/globalModal/index.tsx [hunk 2]
```diff
@@ -234,10 +234,12 @@ const Modal = styled(motion.div)`
   max-width: 100%;
   width: 640px;
   pointer-events: auto;
-  padding: 80px ${space(1.5)} ${space(2)} ${space(1.5)};
+  margin-top: 64px;
+  padding: ${space(2)} ${space(1.5)} ${space(2)} ${space(1.5)};
```
Comment (vuluongj20) [inline static/app/components/globalModal/index.tsx:238]: We can shorten this to `padding: ${space(2)} ${space(1.5)};`

File: static/app/components/globalModal/index.tsx [hunk 3]
```diff
@@ -234,10 +234,12 @@ const Modal = styled(motion.div)`
...
 
   @media (min-width: ${p => p.theme.breakpoints.medium}) {
-    padding: 80px ${space(2)} ${space(4)} ${space(2)};
+    margin-top: 50px;
+    padding: ${space(4)} ${space(2)} ${space(4)} ${space(2)};
```
Comment (vuluongj20) [inline static/app/components/globalModal/index.tsx:242]: Same as above: `padding: ${space(4)} ${space(2)};`

**Review by vuluongj20 — APPROVED** [review]
Looks good! Thanks for the PR!

There's still a small padding area around the dialog that's not clickable. I'm not sure why it's there, but we can probably remove it and turn it all into margins. Doing so would affect the dialog's width though, due to box-sizing, so let's not do that just yet.
